### PR TITLE
bump(main/ncdu2): 2.6

### PR DESCRIPTION
**Pull request by TomJo2000**
`ncdu` now depends on `zstd` as of version 2.6[^1]
I've taken the opportunity to refactor the `post_get_source` step.

A minor patch to the Makefile is also required to ensure it respected the set `${STRIP}` program.

[^1]: upstream changelog https://g.blicky.net/ncdu.git/tree/ChangeLog?h=v2.6#n6

### CODE REVIEW

**Comment by TomJo2000** [comment]
Works on device.
We may wanna consider replacing the references to `ncdu` in the `--help` output and man page with `ncdu2` since we rename the binary so `ncdu2` can be installed alongside `ncdu`.

The new features such as multithreaded scanning with `-t` work without issue.

**Comment by TomJo2000** [comment]
Replaced all relevant references to `ncdu` with `ncdu2`,
and included the `ncdubinexp.pl` utility for reading, validating and analyzing `ncdu` export files.
(renamed to `ncdu2binexp` as the format differs between `ncdu` 1.x and 2.x)

I can split out `fix-progname.patch` into multiple smaller patches if desired.

File: packages/ncdu2/fix-progname.patch [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+@@ -492,14 +492,14 @@ pub fn main() void {
+     };
+ 
+     if (loadConf) {
+-        tryReadArgsFile("/etc/ncdu.conf");
++        tryReadArgsFile("@TERMUX_PREFIX@/etc/ncdu2.conf");
+ 
+         if (std.posix.getenvZ("XDG_CONFIG_HOME")) |p| {
+-            const path = std.fs.path.joinZ(allocator, &.{p, "ncdu", "config"}) catch unreachable;
++            const path = std.fs.path.joinZ(allocator, &.{p, "ncdu2", "config"}) catch unreachable;
```
Comment (TomJo2000) [inline packages/ncdu2/fix-progname.patch:206]: > [!IMPORTANT]
> This change does modify the config location for `ncdu2` which may be undesirable.

I have taken care to also modify the corresponding section of the man page,
but changing the config location from `XDG_CONFIG_HOME/ncdu/config` to `XDG_CONFIG_HOME/ncdu/config` may still cause undue confusion to users.

The alternative would be sharing the config location between the two `ncdu`'s,
which may lead to incompatibilities.
<sub>Although who's realistically gonna use both?</sub>

Comment (fornwall) [reply packages/ncdu2/fix-progname.patch:206]: I wonder if it's worth the trouble and potential confusion to do the `ncdu2` rename at all, and if we instead can let this package conflict with `TERMUX_PKG_CONFLICTS="ncdu"`, so only one can be installed at the same time?

Comment (TomJo2000) [reply packages/ncdu2/fix-progname.patch:206]: Is there even a feature difference between `ncdu` and `ncdu2` at this point?<sub>(rhet. question)</sub>
We may as well just drop `ncdu` 1.x and replace it with 2.x if there is not.

Comment (truboxl) [reply packages/ncdu2/fix-progname.patch:206]: The zig based packages often have a minimum kernel requirement https://github.com/termux/termux-packages/issues/20783#issuecomment-2211719239. I personally havent verified ncdu2 on old kernel. I expect C based 1.x can keep working for really old devices.

File: packages/ncdu2/build.sh
Comment (truboxl) [inline packages/ncdu2/build.sh:1]: I think can move the existing Makefile patching (line 46) out of build.sh

Comment (TomJo2000) [reply packages/ncdu2/build.sh:1]: Good catch.

File: packages/ncdu2/fix-progname.patch (hunk 1 above)
Comment (TomJo2000) [reply packages/ncdu2/fix-progname.patch:206]: How old are we talking here?
Is there any official documentation from Zig on the minimum supported Linux kernel version?

File: packages/ncdu2/build.sh
Comment (TomJo2000) [reply packages/ncdu2/build.sh:1]: I moved the `--with-default-terminfo-dir` path fix into the patch and renamed it from `Makefile.patch` to `fix-hardcoded-paths.patch`
It also includes the fix to the config file location in `main.zig`
Which has accordingly been taken out of `fix-progname.patch`

File: packages/ncdu2/fix-hardcoded-paths.patch [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+index ad9d7f7..73210ba 100644
+--- a/Makefile
++++ b/Makefile
+@@ -71,8 +71,8 @@ static-%.tar.gz:
+ 		--with-pkg-config-libdir="`pwd`/../inst/pkg"\
+ 		--without-cxx --without-cxx-binding --without-ada --without-manpages --without-progs\
+ 		--without-tests --enable-pc-files --without-pkg-config --without-shared --without-debug\
+-		--without-gpm --without-sysmouse --enable-widec --with-default-terminfo-dir=/usr/share/terminfo\
+-		--with-terminfo-dirs=/usr/share/terminfo:/lib/terminfo:/usr/local/share/terminfo\
++		--without-gpm --without-sysmouse --enable-widec --with-default-terminfo-dir=@TERMUX_PREFIX@/usr/share/terminfo\
```
Comment (TomJo2000) [inline packages/ncdu2/fix-hardcoded-paths.patch:11]: And that's an error...
Left an extra `/usr` in the path.

**Comment by truboxl** [comment]
I can't seem to get ncdu2binexp working
```
Can't locate CBOR/XS.pm in @INC (you may need to install the CBOR::XS module) (@INC entries checked: /data/data/com.termux/files/usr/lib/perl5/site_perl/5.38.2/aarch64-android /data/data/com.termux/files/usr/lib/perl5/site_perl/5.38.2 /data/data/com.termux/files/usr/lib/perl5/5.38.2/aarch64-android /data/data/com.termux/files/usr/lib/perl5/5.38.2) at /data/data/com.termux/files/usr/bin/ncdu2binexp line 29.
BEGIN failed--compilation aborted at /data/data/com.termux/files/usr/bin/ncdu2binexp line 29.
```

**Comment by TomJo2000** [comment]
Wrote a `termux_pkg_auto_update()` function for both `ncdu` and `ncdu2`.
Looks like it still needs some tweaking for `ncdu`...
For whatever it thinks `1.9 >= 1.19`

**Comment by TomJo2000** [comment]
Hmm the auto update functions should probably be a separate commit.

File: packages/ncdu2/fix-progname.patch (hunk 1 above)
Comment (TomJo2000) [reply packages/ncdu2/fix-progname.patch]: I think deliberating whether to keep both `ncdu` packages around can be left for another time.
For now, if we're keeping both around I think they should have separate config files, and the `--help` output and man page for `ncdu2` should mention `ncdu2` as the program name.

**Pull request merged**